### PR TITLE
avoid out-of-memory serving large logfiles

### PR DESCRIPTION
**Pull request by divinity76**
large logfiles previously resulted in out-of-memory errors, see https://github.com/hestiacp/hestiacp/issues/2736

### CODE REVIEW

**Comment by divinity76** [comment]
PS this code should be updated again if/when PR #2720 gets merged

**Comment by jaapmarcus** [comment]
Call to undefined function passthru() in /usr/local/hestia/web/download/web-log/index.php:36

**Comment by divinity76** [comment]
@jaapmarcus really? that makes no sense! it's a php built-in function! what do you get from `var_dump(ini_get("disable_functions"));`  ?

**Comment by jaapmarcus** [comment]
pcntl_alarm,pcntl_fork,pcntl_waitpid,pcntl_wait,pcntl_wifexited,pcntl_wifstopped,pcntl_wifsignaled,pcntl_wifcontinued,pcntl_wexitstatus,pcntl_wtermsig,pcntl_wstopsig,pcntl_signal,pcntl_signal_get_handler,pcntl_signal_dispatch,pcntl_get_last_error,pcntl_strerror,pcntl_sigprocmask,pcntl_sigwaitinfo,pcntl_sigtimedwait,pcntl_exec,pcntl_getpriority,pcntl_setpriority,pcntl_async_signals,pcntl_unshare,passthru,system,popen,show_source,

**Comment by jaapmarcus** [comment]
Currently editing the fpm.conf and see if it works

**Comment by myvesta** [comment]
My 2 cents: there is no point to disable passthru() if exec() or shell() is enabled, for example.

**Comment by jaapmarcus** [comment]
> My 2 cents: there is no point to disable passthru() if exec() or shell() is enabled, for example.

I agree how ever it seem to be the default settings in php8.1 probally?

`pcntl_alarm,pcntl_fork,pcntl_waitpid,pcntl_wait,pcntl_wifexited,pcntl_wifstopped,pcntl_wifsignaled,pcntl_wifcontinued,pcntl_wexitstatus,pcntl_wtermsig,pcntl_wstopsig,pcntl_signal,pcntl_signal_get_handler,pcntl_signal_dispatch,pcntl_get_last_error,pcntl_strerror,pcntl_sigprocmask,pcntl_sigwaitinfo,pcntl_sigtimedwait,pcntl_exec,pcntl_getpriority,pcntl_setpriority,pcntl_async_signals,pcntl_unshare,passthru,system,popen,show_source,`

And disable_functions was not set

**Comment by divinity76** [comment]
i don't think it's the default for the official php builds, but it may be the default for deb.sury.org builds? (which hestiacp is using)

**Comment by jaapmarcus** [comment]
https://github.com/hestiacp/hestiacp/blob/18adc9541067a5a4d734fae54be86dc6da56fca1/src/deb/php/php.ini#L311-L312

Found the issue I hope

**Comment by divinity76** [comment]
as @myvesta pointed out, might as well get rid of pcntl_exec, passthru, system, popen while you're at it, having just 1 of these functions enabled is practically as insecure as having all of them enabled ^^

**Comment by jaapmarcus** [comment]
Changed the php.ini as suggested we will need to rebuild after we are releasing it

File: web/download/web-log/index.php [hunk 1]
```diff
@@ -27,10 +27,15 @@
...
 }
-
-exec(HESTIA_CMD."v-list-web-domain-".$type."log $user ".$v_domain." 5000", $output, $return_var);
-if ($return_var == 0) {
-    foreach ($output as $file) {
-        echo $file . "\n";
-    }
+$cmd = implode(" ",array(
+    escapeshellarg(HESTIA_CMD."v-list-web-domain-".$type."log"),
+    escapeshellarg($user),
```
Comment (jaapmarcus) [inline web/download/web-log/index.php]: Both $user and $domain are allready escapeshellarg causing 

 ''\''jaap'\''' ''\''domain.com'\''' 
 
 https://github.com/hestiacp/hestiacp/blob/18adc9541067a5a4d734fae54be86dc6da56fca1/web/download/web-log/index.php#L8-L9
 
 And 

https://github.com/hestiacp/hestiacp/blob/18adc9541067a5a4d734fae54be86dc6da56fca1/web/inc/main.php#L121-L129

**Review by jaapmarcus — CHANGES_REQUESTED** [review]

File: web/download/web-log/index.php (hunk 1 above)
Comment (divinity76) [reply web/download/web-log/index.php]: thanks, i'll check it out

Comment (divinity76) [reply web/download/web-log/index.php]: does this look right? https://github.com/hestiacp/hestiacp/pull/2741/commits/842d114e7756bd5d20184d2a5b2e6422554d9a0a

**Review by jaapmarcus — APPROVED** [review]

**Pull request merged**